simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/Station.java

```diff
@@ -69,7 +69,7 @@ public Station(SimulatorConfiguration.StationConfiguration configuration, OkHttp
         this.configuration = configuration;
         this.state = new StationState(configuration);
 
-        this.webSocketClient = new WebSocketClient(stationMessageInbox, configuration.getId(), new OkHttpWebSocketClient(okHttpClient));
+        this.webSocketClient = new WebSocketClient(this, new OkHttpWebSocketClient(okHttpClient));
```

Comment:
I don't like the idea of giving full access to station here

Reply:
The point is to encapsulate internal fields of the station. All interactions should be performed via public API. I think its a valid argument.

Reply:
Well, Station#updateHeartbeat is not supposed to be called by WebSocketClient, also in scope of https://myevbox.atlassian.net/browse/SPE-687 station will have to re-establish websocket connection. It means that station controls websocket and not vise-versa. Public interface is only for station handlers, otherwise why do we have queue at all?

Reply:
WebSocketClient might call any public API if there is a need. If there is not need to call updateHeartbeat -- then dont call it.
The main point of having a queue is to have simple implementation which is thread-safe. Otherwise you will always have to think of correctness of your code as you will have multi-threaded access (OCPP server & command line).
Queue, State, Configuration is an internal part of the station. Before we were injecting them to the handler-classes, web-socket classes... doing that code becomes less readable, I think we should interact with the station through public API and not through internals. What do you think?

Reply:
If WebSocket will start calling Station via public API we lose thread-safety. There will be a possibility that station state will be modified by station-worker and websocket-worker threads simultaneously.

Reply:
True. Lets why in WebSocket layer as of now - you should call sendMessage method only. 
Another example is: what if we decide to change stationMessageInbox to something else and you will have to change it also in WebSocketLayer unless if it is encapsulated.

I understand what you are saying, that you might call station directly but looks like we not that bad there, for example you can not change the state from websocket layer as its: return StationState.copyOf(state);

Do you disagree with my points? Perhaps updateHeartbeat -- should also be a handler-class.

Reply:
Decided to leave as it is.